packages/demo-protobuf/src/demo.spec.ts

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { assert } from "@cosmjs/utils";
+import path from "path";
+import protobuf from "protobufjs";
+
+import reflectionRoot from "./demo";
+import demoJson from "./demo.json";
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+type MsgDemo = {
+  readonly example: string;
+};
+
+const { Coin } = cosmosSdk.v1;
+const { TxBody } = cosmosSdk.tx.v1;
+const { MsgSend } = cosmosSdk.x.bank.v1;
+const { Any } = google.protobuf;
+
+function pendingWithoutNode(): void {
+  if ((global as any).jasmine) {
+    return pending("Test not enabled for browser");
+  }
+}
+
+function getTypeName(typeUrl: string): string {
+  const parts = typeUrl.split(".");
+  return parts[parts.length - 1];
+}
+
+describe("protobuf demo", () => {
+  it("works with generated static code", () => {
+    const coin = Coin.create({
+      denom: "ucosm",
+      amount: "1234567890",
+    });
+    const msgSend = MsgSend.create({
+      fromAddress: Uint8Array.from(Array.from({ length: 20 }, () => 1)),
+      toAddress: Uint8Array.from(Array.from({ length: 20 }, () => 2)),
+      amount: [coin],
+    });
+    const msgSendBytes = MsgSend.encode(msgSend).finish();
+    const msgSendWrapped = Any.create({
+      type_url: "cosmos_sdk/cosmos.bank.MsgSend",
+      value: msgSendBytes,
+    });
+    const txBody = TxBody.create({
+      messages: [msgSendWrapped],
+      memo: "Some memo",
+      timeoutHeight: 9999,
+      extensionOptions: [],
+    });
+    const txBodyBytes = TxBody.encode(txBody).finish();
+
+    // Deserialization
+    const txBodyDecoded = TxBody.decode(txBodyBytes);
+    const msg = txBodyDecoded.messages[0];
+    assert(msg.value);
+    const msgSendDecoded = MsgSend.decode(msg.value);
+
+    // fromAddress and toAddress are now Buffers
+    expect(Uint8Array.from(msgSendDecoded.fromAddress)).toEqual(msgSend.fromAddress);
+    expect(Uint8Array.from(msgSendDecoded.toAddress)).toEqual(msgSend.toAddress);
+    expect(msgSendDecoded.amount).toEqual(msgSend.amount);
+  });
+
+  it("works with dynamically loaded proto files", async () => {
+    pendingWithoutNode();
+    const protoPath = path.join(__dirname, "demo.proto");
+    const root = await protobuf.load(protoPath);
```

Comment:
Here is a code snippet that shows how to load .proto files in a browser. I created that as part of a very very ealy demo locally:

```ts
const source = `
package awesomepackage;
syntax = "proto3";

import "google/protobuf/timestamp.proto";

message Person {
  string first_name = 1;
  string last_name = 2;
}

message AwesomeMessage {
  uint32 awesome_number = 2; // becomes awesomeNumber
  string title = 1;
  google.protobuf.Timestamp created = 3;
  google.protobuf.Any message = 5;
  Person author = 6;
  bytes checksum = 100;
  string description = 4;
}
`;

const root = proto
  .parse(source)
  .root.addJSON(proto.common.get("google/protobuf/timestamp.proto")!.nested!)
  .addJSON(proto.common.get("google/protobuf/any.proto")!.nested!)
  .resolveAll();
// const root = parsed.root;

// Obtain a message type
const AwesomeMessage = root.lookupType("awesomepackage.AwesomeMessage");
```